### PR TITLE
Fix scheduler; Add config kwargs; Add s1K

### PR DESCRIPTION
**Pull request by Muennighoff**
Explanation for changes:
- accelerate steps the processor once for each process so we can either turn that off (implemented solution) or increase the num training steps by num processes; If leaving it as is, what happens is that when running on 2 processes like in the below img it will increase the LR again after reaching 0 --- note that if merging this PR the LR scheduling in your current runs will change
Screenshot 2025-03-03 at 2 14 13 PM

With e.g. 16 processes on s1K-1.1 it looks like below 
Screenshot 2025-03-03 at 2 20 50 PM

- Without the * epochs, the scheduler will also reset after each epoch a bit like above imgs
- Current code logs on examples which is a bit confusing in wandb cuz wandb assumes that the x axis is # steps; E.g. with the current code the plot above would actually be 500 steps; with the changes here it is the correct step count; the downside is that if using grad acc it relogs at the same count but maybe that is fine or can add an if condition to just not log during grad acc (only at the end)
- The new config.yaml values are the default ones so default behavior won't change

### CODE REVIEW

**Comment by kawine** [comment]
Great catch on all of these!

**Comment by Muennighoff** [comment]
Okay just be careful with your current runs as the scheduling may be different now!

**Comment by kawine** [comment]
I verified that things still work with `step_scheduler_with_optimizer` set to False, although the warmup period has to be roughly doubled; otherwise there is some instability. Also fixed some bugs in the calculation of total steps for the other Trainers, similar to what you did for SFTTrainer (thanks!). Just pushed those changes.
